extensions/integration-tests/testfmks/src/stress.ts

```diff
@@ -0,0 +1,289 @@
+		trace(`parameters to Stress constructor: runtime=${runtime}, dop=${dop}, iterations=${iterations}, passThreshold=${passThreshold}`);
+		trace(`default properties of this Stress object: this.runtime=${this.runtime}, this.dop=${this.dop}, this.iterations=${this.iterations}, this.passThreshold=${this.passThreshold}`);
+		let x: number;
+		x = this.nullCoalesce(runtime, this.runtime); this.runtime = x;
+		x = this.nullCoalesce(dop, this.dop); this.dop = x;
+		x = this.nullCoalesce(iterations, this.iterations); this.iterations = x;
+		x = this.nullCoalesce(passThreshold, this.passThreshold); this.passThreshold = x;
+
+		// validate this object
+		//
+		validate(this).then(errors => {
+			if (errors.length > 0) {
+				errors.map(error => { throw error; });
+				debug(`validation error in stress object: ${JSON.stringify(errors, undefined, '\t')}`);
+				throw errors;
+			}
+		}).catch(fatalErrors => {
+			if (fatalErrors.length > 0) {
+				fatalErrors.map(error => { throw error; });
+				debug(`fatal error while validating stress object: ${JSON.stringify(fatalErrors, undefined, '\t')}`);
+				throw fatalErrors;
+			}
+		});
+
+		trace(`properties of Stress Object post full construction with given parameters: this.runtime=${this.runtime}, this.dop=${this.dop}, this.iterations=${this.iterations}, this.passThreshold=${this.passThreshold}`);
+	}
+
+	private nullCoalesce(value: number, defaultValue: number): number {
+		return (value === null || value === undefined || isNaN(value)) ? defaultValue : value;
+	}
+
+	/**
+	 *
+	 * @param originalMethod - The reference to the originalMethod that is being stressfied.The name of this method is {@link functionName}
+	 * @param originalObject - The reference to the object on which the {@link originalMethod} is invoked.
+	 * @param functionName - The name of the originalMethod that is being stressfied.
+	 * @param args - The invocation argument for the {@link originalMethod}
+	 * @param runtime - The desconstructed {@link StressOptions} parameter. see {@link StressOptions} for details.
+	 * @param dop - The desconstructed {@link StressOptions} parameter. see {@link StressOptions} for details.
+	 * @param iterations - The desconstructed {@link StressOptions} parameter. see {@link StressOptions} for details.
+	 * @param passThreshold - The desconstructed {@link StressOptions} parameter. see {@link StressOptions} for details.
+	 *
+	 * @returns - {@link StressResult}.
+	 */
+	async run(
+		originalMethod: Function,
+		originalObject: any,
+		functionName: string,
+		args: any[],
+		{ runtime, dop, iterations, passThreshold }: StressOptions = DefaultStressOptions
+	): Promise<StressResult> {
+		// TODO support for cutting of the iterator if runtime has exceeded needs to be implemented.
```

Comment:
If this is true, then we should modify the comment above that this actually has been implemented

Reply:
May be i need to rewrite the comment, but I will just go ahead and implement the runtime parameter.